### ISSUE TITLE
(Lunch) breaks

### ISSUE THREAD

**Issue by N-Wouda** [PyVRP/PyVRP#352]
#### Is your feature request related to a problem? Please describe

Not really, as in, it's not a bug. But I'd like to have some support for (scheduled) breaks, like a lunch break at a fixed time. In short, I'd like us to implement a way to solve the VRP with lunch breaks (VRPLB), for example as described by [Coelho et al. (2016)](https://www.tandfonline.com/doi/abs/10.1057/jors.2015.90?journalCode=tjor20). Ideally we'd do a little more, and also allow the break to be had at a fixed location, rather than just along the route at whatever client's currently being visited.

#### Describe alternatives you have considered

If the breaks are held at a fixed location, it is somewhat possible to do this by creating a dummy client representing the lunch break for each vehicle, and setting the time windows such that they must be assigned to different routes. I already tried that, but PyVRP has a pretty hard time making that work: it takes a while to reach feasibility, and even then, the routes are not particularly good. So that way of making this work is probably not the best.

If the breaks are held along the route, that approach could still work (dummy node with zero distance/duration to all other nodes), but that'd not solve the issue that PyVRP struggles to schedule this.

So, a key part of making this work is explicitly telling PyVRP that the breaks are part of the vehicle type, and _must_ be part of a route.

**Comment by N-Wouda** [comment]
There's also a whole literature on time-based breaks, e.g. after N hours of driving. That gets really complex quick, so I don't think we need to consider that within this issue. Fixed breaks is a good start.

**Comment by N-Wouda** [comment]
### Break at a location

Behaves like a node, since there is travel to/from that location to take the break (service duration), starting between some time windows. This is straightforward to add in: the nodes need to be part of the vehicle's route, and cannot be moved elsewhere. They're a bit like the start and end depots in that sense.

### Breaks along the route

Such a break is had before or after a client service. It might be easiest to do if we somehow add the concept of virtual nodes, places that going to/from takes zero time. Then they could still be nodes in the current sense, but any distance/duration from/to them is automatically 0. But that's a pretty significant departure from the current codebase, so I'm not sure.

---

In any case: breaks at locations is really easy, breaks along the route is a bit trickier. I only need breaks at locations for my own use case, but I'd like to try and support this generally if it's not too difficult to do.

**Comment by N-Wouda** [comment]
Pragmatically, I'll probably start with breaks at a location. We can later adapt that to also support breaks along the route, but we'll first need a basis for that anyway.

**Comment by N-Wouda** [comment]
OK, so, in my application, the break location is pretty close to the depot. I can also model returns for breaks by implementing shift durations on the vehicle types, duplicating the vehicles for each time segment, and adjusting the shift time windows to fall around the breaks. That's close enough, and a lot simpler than breaks, which are turning into a right mess in #364. So let's not do that, and do the shift time windows instead.

**Comment by shubham441996** [comment]
Hi @N-Wouda 

I am trying to model **driver breaks** using the existing **shift feature** in PyVRP. Here’s how I’ve broken down the problem:  

- Vehicle shift: **08:00 – 18:00**  
- Break: **12:30 – 14:00**  

To represent this, I created two vehicles:  

- **Vehicle 1**: 08:00 – 12:30 (Depot → Fake Depot)  
- **Vehicle 2**: 14:00 – 18:00 (Fake Depot → Depot)  

I added a **Fake Depot** with zero distance and duration to/from all orders, since I don’t know where Vehicle 1 should end and where Vehicle 2 should start. The idea is that after optimisation, Vehicle 1 and Vehicle 2 should effectively represent the same physical vehicle before and after the break.  

However, I am running into a few issues with this approach:  

1. **Multiple vehicle types**: When there are several vehicles, the optimisation produces unoptimised routes because Vehicle 1 and Vehicle 2 are treated as independent vehicles, leading to overlaps.  
2. **Capacity constraints**: It’s unclear how to assign capacity between Vehicle 1 and Vehicle 2. For example, if 90% of orders fall before the break and only 10% after, splitting capacity between them becomes problematic.  
3. **Merging routes**: After optimisation, Vehicle 1’s end location should become Vehicle 2’s start location, but I’m not sure how to adjust timing and routing constraints to ensure this works correctly.  

**Question:**  
Is there a better way in PyVRP to model breaks (within a shift) without splitting a single vehicle into two and using a fake depot? Or is there a recommended approach to handle the above issues when using this workaround?
